### PR TITLE
Add CI (github actions); add third_party/python/markupsafe.

### PR DESCRIPTION
**Pull request by tcal-x**
CI runs two jobs in parallel -- check-docs and check-sw.    "check-sw" compiles the software for `proj/mnv2_first`, requiring that it goes through all of the CFU-Playground setup and also manually installs a good RISC-V toolchain.

Signed-off-by: Tim Callahan <tcal@google.com>

### CODE REVIEW

**Comment by tcal-x** [comment]
check-docs is redundant since a PR also triggers a doc build on readthedocs.   I will probably remove it.

File: scripts/setup-ci [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+fi
+
+if [ ${#missing[@]} -gt 0 ]; then
+    echo "Missing packages: ${missing[@]}" >&2
+    sudo apt install -y "${missing[@]}"
+fi
+
+# Warn about different versions
+
+if ! riscv64-unknown-elf-gcc --version | grep ' 8.3.' >/dev/null; then
```
Comment (alanvgreen) [inline scripts/setup-ci:34]: The toolchain I'm using has this version:

riscv64-unknown-elf-gcc (SiFive GCC 10.1.0-2020.08.2) 10.1.0

Maybe we need to update scripts/setup too.

File: .github/workflows/sample.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: sample
```
Comment (alanvgreen) [inline .github/workflows/sample.yml:1]: A more descriptive name? (same for filename)

File: scripts/setup-ci [hunk 3]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
```
Comment (alanvgreen) [inline scripts/setup-ci:1]: Can this file be integrated with scripts/setup?

It would be a pity if they fell out of sync.

**Review by alanvgreen — APPROVED** [review]
Am happy for this to go in as-is, and these comments can be addresed later.

File: scripts/setup-ci (hunk 1 above)
Comment (tcal-x) [reply scripts/setup-ci:34]: Oh, so 8.3 is good, 9.x is bad, but 10.1 is good again?   I'll have to check that out.

File: .github/workflows/sample.yml (hunk 2 above)
Comment (tcal-x) [reply .github/workflows/sample.yml]: Yep, done.

**Pull request merged**